src/main/java/io/github/syst3ms/skriptparser/util/math/NumberMath.java

```diff
@@ -200,6 +202,41 @@ public static Number tanh(Number n) {
         }
     }
 
+    /**
+     * Checks if a given BigInteger is prime. The certainty of this check is always 100%
+     * @param number the number to check
+     * @return whether or not this number is a prime
+     */
+    public static boolean isPrime(BigInteger number) {
+        // 0. Check if the number is of the form 6x±1 for values higher than 3.
+        if (!number.subtract(BigInteger.ONE).mod(BigInteger.valueOf(6)).equals(BigInteger.ZERO)
+                && !number.add(BigInteger.ONE).mod(BigInteger.valueOf(6)).equals(BigInteger.ZERO)
+                && number.compareTo(BigInteger.valueOf(3)) > 0)
+            return false;
+
+        // 1. Check in the cached primes.
+        if (cachedPrimes.contains(number))
+            return true;
+
+        // 2. Check via BigInteger.isProbablePrime(certainty)
+        // If this returns false, it is definitely not a prime.
+        if (!number.isProbablePrime(5))
+            return false;
+
+        // 3. If the number is not too high, we can compute new primes and add them to the cached list.
+        if (number.compareTo(BigInteger.valueOf(10_000_000)) < 0) {
```

Comment:
This criterion is far too lax. Sure, we don't want to compute primes up to 10 million, but the 1000th prime is only 7919, and the largest prime below 10 million is 9,999,991, and that is the 664,579th prime number. If someone checked the primality of a number around that size, this would end up computing around 663,000 primes, which is FAR TOO MUCH.

Instead, we want to impose a bound on how many prime numbers we want to generate at once. I am bad at evaluating performance, so I'll give you the choice of that number, but I'll do the math to determine how to effectively check this condition.

Let k be the number of primes generated so far, n the number whose primality we want to check, and A the maximum amount of primes we want to generate in a single run. Then a lower bound on the number of primes below n is given by n/log(n), where log(n) is the natural logarithm. Then our condition can be expressed as :
n/log(n)-k < A
Now, a small optimisation that can be used instead of computing a (costly) log right away : if n is sufficiently close to the largest prime generated so far (let it be p), we can just generate the primes between and not even think about it since there are few enough. The quantity A might even be a good threshold for such behavior, since if n-p<A, the expected amount of generated primes will be A/log(A) which is definitely smaller than A.